File: app/src/main/java/com/example/testekamba/models/activities.java
package com.example.testekamba.models;

import com.google.gson.annotations.SerializedName;

public class activities {


    @SerializedName("transaction_type")
    private String transaction_type;

    @SerializedName("id")
    private  String id_operation;

    @SerializedName("amount")
    private String amount;

    private String subtotal;

    private String fee;

    @SerializedName("created_at")
    private String date;

    @SerializedName("from")
    private from fromUser;

    @SerializedName("to")
    private to toUser;

    private String description;

    private String reference_checkout;

    private String reference_recharge;

    private String status;

    private String intent;
    private String recharge_code;
    private String echarge_price;
    private String amount_of_utts;
    private String validity_days;
    private String mobile_operator_name;
    private String mobile_operator_type;
    private String amount_paid;

    private String id_mobile_operator_type;
    private String recharge_price;

    public String getRecharge_price() {
        return recharge_price;
    }

    public String getId_mobile_operator_type() {
        return id_mobile_operator_type;
    }

    public String getRecharge_code() {
        return recharge_code;
    }

    public String getEcharge_price() {
        return echarge_price;
    }

    public String getAmount_of_utts() {
        return amount_of_utts;
    }

    public String getValidity_days() {
        return validity_days;
    }

    public String getMobile_operator_name() {
        return mobile_operator_name;
    }

    public String getMobile_operator_type() {
        return mobile_operator_type;
    }

    public String getAmount_paid() {
        return amount_paid;
    }

    @SerializedName("payment_method")
    private payment_method payment_method;
    @SerializedName("company_bank_account")
    private bank_account bank_account;


    public activities.bank_account getBank_account() {
        return bank_account;
    }

    public String getReference_checkout() {
        return reference_checkout;
    }

    public String getReference_recharge() {
        return reference_recharge;
    }

    public String getStatus() {
        return status;
    }

    public String getIntent() {
        return intent;
    }

    public String getTransaction_type() {
        return transaction_type;
    }

    public String getId_operation() {
        return id_operation;
    }

    public String getAmount() {
        return amount;
    }

    public String getSubtotal() {
        return subtotal;
    }

    public String getFee() {
        return fee;
    }

    public String getData() {
        return date;
    }

    public from getFromUser() {
        return fromUser;
    }


    public to getToUser() {
        return toUser;
    }

    public String getDescription() {
        return description;
    }

    public String getDate() {
        return date;
    }

    public payment_method getPayment_method() {
        return payment_method;
    }

    public class from
    {
        @SerializedName("id")
        public  String from_id;

        public String firstname;

        public String lastname;

        public long phone;

        public  String email;

        public String profile_photo;


    }

    public class to
    {
        @SerializedName("id")
        public  String from_id;

        public String firstname;

        public String lastname;

        public long phone;

        public  String email;

        public String profile_photo;


    }

    public class payment_method
    {
        public   String id;
        public  String name;
        public  String description;

    }

    public class bank_account
    {
        public   String id;
        public  String iban;
        public  String number;
        public  String typeCof;
        @SerializedName("holder")
        public  String holder;
        public  bank bank;
    }


    public  class  bank{
        public   String id;
        public  String name;
        public  String description;
    }



}

File: teste_alpha/app/src/main/java/com/filomenadeveloper/teste_alpha/models/Activities.kt
package com.filomenadeveloper.teste_alpha.models

import com.google.gson.annotations.SerializedName

data class Activities(
        @SerializedName("id") val id:String,
        @SerializedName("intent") val intent:String,
        @SerializedName("amount") val amount:Float,
        @SerializedName("subtotal") val subtotal:Float,
        @SerializedName("fee") val fee:Float,
        @SerializedName("bank_account") val bank_account: Bank_Account,
        @SerializedName("from") val from: From,
        @SerializedName("to") val to: To,
        @SerializedName("payment_method") val payment_method: Payment_Method,
        @SerializedName("company_bank_account") val company_bank_account: Company_Bank_Account,
        @SerializedName("phone_number") val phone_number: String,
        @SerializedName("status") val status: String,
        @SerializedName("committed") val committed: Boolean,
        @SerializedName("paid") val paid: String,
        @SerializedName("sms_recharge_scheme") val sms_recharge_scheme: String,
        @SerializedName("phone_number_service") val phone_number_service: String,
        @SerializedName("mobile_recharge_card_id") val mobile_recharge_card_id: String,
        @SerializedName("transaction_id") val transaction_id: String,
        @SerializedName("recharge_code") val recharge_code: String,
        @SerializedName("recharge_price") val recharge_price: String,
        @SerializedName("amount_of_utts") val amount_of_utts: String,
        @SerializedName("validity_days") val validity_days: String,
        @SerializedName("mobile_operator_name") val mobile_operator_name: String,
        @SerializedName("mobile_operator_type") val mobile_operator_type: String,
        @SerializedName("amount_paid") val amount_paid: String,
        @SerializedName("created_at") val created_at: String,
        @SerializedName("ussd_recharge_scheme") val ussd_recharge_scheme: String,
        @SerializedName("serial_number") val serial_number: String,
        @SerializedName("description") val description: String,
        @SerializedName("transaction_type") val transaction_type: String,
        @SerializedName("reference_checkout") val reference_checkout: String,
        @SerializedName("reference_recharge") val reference_recharge: String,
        @SerializedName("receipt_image") val receipt_image: String,
        @SerializedName("expires_at") val expires_at: String

    ) {
}

